server/entity/splash_potion.go

```diff
@@ -0,0 +1,248 @@
+package entity
+
+import (
+	"github.com/df-mc/dragonfly/server/block/cube"
+	"github.com/df-mc/dragonfly/server/entity/effect"
+	"github.com/df-mc/dragonfly/server/entity/physics"
+	"github.com/df-mc/dragonfly/server/entity/physics/trace"
+	"github.com/df-mc/dragonfly/server/internal/nbtconv"
+	"github.com/df-mc/dragonfly/server/item/potion"
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/df-mc/dragonfly/server/world/particle"
+	"github.com/df-mc/dragonfly/server/world/sound"
+	"github.com/go-gl/mathgl/mgl64"
+	"image/color"
+	"time"
+)
+
+// SplashPotion is an item that grants effects when thrown.
+type SplashPotion struct {
+	transform
+	yaw, pitch float64
+
+	ticksLived int
+
+	closeNextTick bool
+
+	owner world.Entity
+
+	t potion.Potion
+	c *ProjectileComputer
+}
+
+// NewSplashPotion ...
+func NewSplashPotion(pos mgl64.Vec3, yaw, pitch float64, owner world.Entity, t potion.Potion) *SplashPotion {
+	s := &SplashPotion{
+		yaw:   yaw,
+		pitch: pitch,
+		owner: owner,
+
+		t: t,
+		c: &ProjectileComputer{&MovementComputer{
+			Gravity:           0.05,
+			Drag:              0.01,
+			DragBeforeGravity: true,
+		}},
+	}
+	s.transform = newTransform(s, pos)
+
+	return s
+}
+
+// Name ...
+func (s *SplashPotion) Name() string {
+	return "Splash Potion"
+}
+
+// EncodeEntity ...
+func (s *SplashPotion) EncodeEntity() string {
+	return "minecraft:splash_potion"
+}
+
+// AABB ...
+func (s *SplashPotion) AABB() physics.AABB {
+	return physics.NewAABB(mgl64.Vec3{-0.125, 0, -0.125}, mgl64.Vec3{0.125, 0.25, 0.125})
+}
+
+// SetType ...
+func (s *SplashPotion) SetType(t potion.Potion) {
```

Comment:
Any reason to expose this method? We should just pass it to the New method. See 5157a8cae26406d6327a21695f6b923900c1f1d9 for what I mean. I think it's better we keep the potion type immutable.